plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -132,7 +132,20 @@ class Pub(
 
     private val flutterAbsolutePath = flutterHome.resolve("bin")
 
-    private val gradleFactory = analyzerConfig.determineEnabledPackageManagers().find { it.type.startsWith("Gradle") }
+    private val gradleFactory = analyzerConfig.determineEnabledPackageManagers()
+        .filter { it.type.startsWith("Gradle") }
```

Comment:
The arbitrary selection, if keeping this at all, should be deterministic. I presume adding `sorted()` would make sense. Maybe preferred: `sortedDescending().first()` or `sorted().last()` ?

Reply:
I'll go for `sorted().last()`, which IDEA proposed to shorten to `maxOf { it }`.